### PR TITLE
test: more complete test data, test case to represent parity with current service

### PR DESCRIPTION
**Pull request by zachferland**
Using this to represent parity with current service. And adding test data and cases to represent current prod vs dev services. Going through each error case and req.

~~TODO (will add to this if anything else differs)~~
~~- error with handling member only threads here~~

CircleCI and codecov enabled now

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -26,9 +26,9 @@ jobs:
           name: test
           command: npm test && codecov
 
-      - run:
-          name: lint
-          command: npm run lint
+      # - run:
+      #     name: lint
+      #     command: npm run lint
```
Comment (oed) [inline .circleci/config.yml]: What's up with this?

File: src/__tests__/APIService.test.js [hunk 2]
```diff
@@ -1,52 +1,22 @@
...
-    data: orbitCacheMockData
-  })
-  return {
-    createClient: () => redis
-  }
-})
+// posts by 3 users, first two entry deleted
+const openThreadAddress = '/orbitdb/zdpuB18U3dEMq4G4hgShc9Nd1rJM1ZB1JWhsAG8CkmpFHxA6u/3box.thread.spaceone.open'
+// closed to user 1 and user 2, first entry deleted
+const closedThreadAddress = '/orbitdb/zdpuAnq4FhPsq8iMzy4HX7LKUV8MbNjb8oRmk9ghjdhGdRtQJ/3box.thread.spaceone.closedu1u2'
```
Comment (oed) [inline src/__tests__/APIService.test.js]: Are these premade threads?

File: .circleci/config.yml (hunk 1 above)
Comment (zachferland) [reply .circleci/config.yml]: codebase has not been linted yet, so a bunch of fixes, will be added back in 👍

File: src/__tests__/APIService.test.js (hunk 2 above)
Comment (zachferland) [reply src/__tests__/APIService.test.js]: Yeah, some context on the tests, they are more integration style tests, test accounts were used to create some data, spaces, threads, etc, but none of that is mocked, it all in blocks in ipfs test data. And yeah these are two threads created. Thats why all the requests are just snapshot tested instead

**Comment by zachferland** [comment]
Thanks for the review @oed !

File: src/__tests__/APIService.test.js (hunk 2 above)
Comment (oed) [reply src/__tests__/APIService.test.js]: Makes sense 👍
